doc/dev/crimson/poseidonstore.rst

```diff
@@ -0,0 +1,254 @@
+===============
+ PoseidonStore
+===============
+
+Key concepts and goals
+======================
+
+* As one of the pluggable backend stores for Crimson, PoseidonStore targets only 
+  high-end NVMe SSDs (not concerned with ZNS devices).
+* Hybrid update strategies for different data types (in-place, out-of-place) to
+  minimize CPU consumption by reducing host-side GC.
+* Utilize NVMe feature (atomic large write command)
+* Sharded data/processing model
+* Support transactions
+
+Background
+----------
+
+- Log-structured store 
+  This kind of store manages a log to store/read the data. Whenever the write is coming 
+  to the store, the data along with metadata is appended at the end of the log. 
+  - Pros
+    - Without a doubt, one sequential write is enough to store the data
+    - It naturally supports transaction (this is no overwrite, so the store can rollback 
+      previous stable state)
+    - Flash friendly (Definately, it mitigates GC burden on SSDs)
+  - Cons
+    - There is host-side GC that induces overheads 
+      - I/O amplification (host-side)
+      - More host-CPU consumption
+    - Slow metadata lookup
+    - Space overhead (live and unused data co-exist)
+
+- In-place update store
+  Conventional filsystems such as ext4, xfs have an in-place update manner. This type of 
+  the store overwrites data and metadta at a fixed offset depending on a file.
+
+  - Pros
+    - No host-side GC is required
+    - Fast lookup
+    - No space overhead
+  - Cons
+    - More writes occur to record the data (metadata and data section are separated)
+    - It cannot support transaction. WAL is required to support transactions
```

Comment:
It *can* support transactions -- using a WAL.  Just replace this line with "Some form of WAL required to ensure update atomicity in the general case."